src/clam/graph.rs

```diff
@@ -120,7 +121,7 @@ impl<T: Number, U: Number> Graph<T, U> {
             indices: vec![],
             depth: 0,
             min_depth: 0,
-            edges_dict: Arc::new(DashMap::new()),
+            edges_dict: Arc::new(HashMap::new()),
```

Comment:
Any reasoning for this transition?

Reply:
DashMaps are mutable by default. I want to expose these as HashMap/HashSet while internally working with DashMap/DashSet for concurrent processing

Reply:
This is the first step to that goal. I think I may have figured out compression but it requires a lot of changes to Cluster so I'm going to finish this transition after I'm done with compression